lib/segment/src/common/rocksdb_buffered_update_wrapper.rs

```diff
@@ -47,17 +51,16 @@ impl DatabaseColumnScheduledUpdateWrapper {
         K: AsRef<[u8]>,
     {
         let key = key.as_ref();
-        // keep `insert_pending_persistence` lock for atomicity
-        let mut insert_guard = self.insert_pending_persistence.lock();
-        insert_guard.remove(key);
-        self.deleted_pending_persistence.lock().insert(key.to_vec());
-        drop(insert_guard);
+        let mut pending_guard = self.pending_operations.lock();
+        pending_guard.inserted.remove(key);
+        pending_guard.deleted.insert(key.to_vec());
         Ok(())
     }
 
     pub fn flusher(&self) -> Flusher {
-        let ids_to_insert = mem::take(&mut *self.insert_pending_persistence.lock());
-        let ids_to_delete = mem::take(&mut *self.deleted_pending_persistence.lock());
+        let pending_operations = mem::take(&mut *self.pending_operations.lock());
+        let ids_to_insert = pending_operations.inserted;
+        let ids_to_delete = pending_operations.deleted;
```

Comment:
Nit: how about struct destructing here?

Reply:
very good point, it will please Andrey :)